### ISSUE TITLE
ShoehornFormatter mysteriously appends whitespace 

### ISSUE THREAD

**Issue by wimglenn** [cjw296/shoehorn#2]
`ShoehornFormatter` is appending whitespace and it's throwing off my asserts .. 

```
>>> from shoehorn.stdlib import ShoehornFormatter
>>> from testfixtures import LogCapture
>>> import logging
>>> with LogCapture() as log:
...     logging.info('blah')
...     
>>> ShoehornFormatter().format(log.records[0])
'blah '
```

I think the issue is that the formatter puts `'%(message)s %(shoehorn_context)s'` even if `shoehorn_context` is empty.

**Comment by cjw296** [comment]
Can you give an example of the kind of asserts you're making?

You're likely hitting this:

https://github.com/cjw296/shoehorn/blob/master/shoehorn/stdlib.py#L52

...specifically `'%(message)s %(shoehorn_context)s'`.

So, somehow we need to get the key/value stuff into a standard library formatted log message. Open to suggestions, but it might be most pragmatic to do your asserts different. If you can point me to them or give some examples, I'd be happy to make suggestions.

**Comment by cjw296** [comment]
Closed by #2.
